gui/mod-manager.lua

```diff
@@ -401,6 +404,104 @@ function ModmanageScreen:init()
     }
 end
 
+ModlistMenu = defclass(ModlistMenu, widgets.Window)
+ModlistMenu.ATTRS {
+    frame_title = "Active Modlist",
+
+    resize_min = { w = 30, h = 15 },
+    frame = { w = 40, h = 15 },
+
+    resizable = true,
+}
+
+local function getWorldModlist(detailed)
+    -- ordered map of mod id -> {handled=bool, versions=map of version -> path}
+    local mods = utils.OrderedTable()
+    local mod_paths = {}
+
+    -- if a world is loaded, process active mods first, and lock to active version
+    if dfhack.isWorldLoaded() then
+        for _,path in ipairs(df.global.world.object_loader.object_load_order_src_dir) do
+            path = tostring(path.value)
+            -- skip vanilla "mods"
+            if not path:startswith(INSTALLED_MODS_PATH) then goto continue end
+            local id, numerical_version, name, steam_id, display_version = scriptmanager.get_mod_info(path)
+            if not id or not numerical_version then goto continue end
+            mods[id]= {handled=true, name=name, version=display_version, steam_id=steam_id}
+            scriptmanager.add_mod_paths(mod_paths, id, path, '.')
+            ::continue::
+        end
```

Comment:
this is breaking the abstraction of the script manager. this script should not have knowledge of INSTALLED_MODS_PATH